File: modules/076-skills-management-system/tools/skills-cli/cmd/list.go
package cmd

import (
	"flag"
	"fmt"
	"os"

	"github.com/vibecoding/skills-cli/internal/config"
	"github.com/vibecoding/skills-cli/internal/gitops"
)

// RunList handles the `skills list` command.
func RunList(args []string) {
	fs := flag.NewFlagSet("list", flag.ExitOnError)
	fs.Usage = func() {
		fmt.Print(`List all available skills in the repository.

Usage:
  skills list

Active skills (checked out in this workspace) are marked with ✅.
Other skills exist in the repo but are not part of your current groups.

`)
		fs.PrintDefaults()
	}

	if err := fs.Parse(args); err != nil {
		os.Exit(1)
	}

	if len(fs.Args()) > 0 && (fs.Args()[0] == "--help" || fs.Args()[0] == "-h") {
		fs.Usage()
		return
	}

	cfg, err := config.Load()
	if err != nil {
		fmt.Fprintln(os.Stderr, "Error:", err)
		os.Exit(1)
	}

	allSkills, err := gitops.ListAllSkills(cfg.RepoPath())
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error: failed to list skills: %v\n", err)
		os.Exit(1)
	}

	// Build a fast lookup set for active skills
	activeSet := make(map[string]bool, len(cfg.Skills))
	for _, s := range cfg.Skills {
		activeSet[s] = true
	}

	fmt.Printf("Skills repository: %s\n", cfg.RepoURL)
	fmt.Printf("Groups:           %v\n\n", cfg.Groups)

	activeCount := 0
	for _, s := range allSkills {
		if activeSet[s] {
			fmt.Printf("  ✅ %s\n", s)
			activeCount++
		} else {
			fmt.Printf("  ○  %s\n", s)
		}
	}

	fmt.Printf("\nActive: %d  |  Total: %d\n", activeCount, len(allSkills))
}

File: modules/076-skills-management-system/tools2/skills-cli/src/commands/list.ts
import * as config from '../lib/config';
import * as gitops from '../lib/gitops';

export function runList(args: string[]): void {
  if (args.includes('--help') || args.includes('-h')) {
    console.log(`List all available skills in the repository.

Usage:
  skills list

Active skills (checked out in this workspace) are marked with ✅.
Other skills exist in the repo but are not part of your current groups.

`);
    return;
  }

  const cfg = config.load();
  const repoDir = config.REPO_SUB_DIR;

  let allSkills: string[];
  try {
    allSkills = gitops.listAllSkills(repoDir);
  } catch (err) {
    console.error(`Error: failed to list skills: ${err}`);
    process.exit(1);
  }

  const activeSet = new Set<string>(cfg.skills);

  console.log(`Skills repository: ${cfg.repo_url}`);
  console.log(`Groups:           ${cfg.groups.join(', ')}\n`);

  let activeCount = 0;
  for (const s of allSkills) {
    if (activeSet.has(s)) {
      console.log(`  ✅ ${s}`);
      activeCount++;
    } else {
      console.log(`  ○  ${s}`);
    }
  }

  console.log(`\nActive: ${activeCount}  |  Total: ${allSkills.length}`);
}
